packages/techdocs-addons/src/test-utils/test-utils.tsx

```diff
@@ -0,0 +1,230 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// import order matters for jest manual mocks! import this first.
+import {
+  useTechDocsReaderDom,
+  useParams,
+  useTechDocsMetadata,
+  useEntityMetadata,
+} from './mocks';
+
+import React, { ReactElement, Fragment } from 'react';
+
+// Shadow DOM support for the simple and complete DOM testing utilities
+// https://github.com/testing-library/dom-testing-library/issues/742#issuecomment-674987855
+import { screen } from 'testing-library__dom';
+import { renderToStaticMarkup } from 'react-dom/server';
+import { Route, Routes } from 'react-router-dom';
+import { act, render } from '@testing-library/react';
+
+import {
+  wrapInTestApp,
+  TestApiProvider,
+  TestApiProviderProps,
+} from '@backstage/test-utils';
+
+import { TechDocsAddons } from '../addons';
+
+/**
+ * @public
+ */
+export type Apis = TestApiProviderProps<any>['apis'];
+
+/**
+ * options for {@link TechDocsAddonBuilder}
+ * @public
+ */
+export type TechDocsAddonBuilderOptions = {
+  dom: ReactElement;
+  entity: any; // TODO: TechDocsEntityMetadata type in techdocs-common?
+  metadata: any; // TODO  TechDocsMetadata in techdocs-common?;
+  componentId: string;
+  readerPage?: ReactElement;
+  apis: Apis;
+  path: string;
+};
+
+const defaultOptions: TechDocsAddonBuilderOptions = {
+  dom: <></>,
+  entity: {},
+  metadata: {},
+  componentId: 'docs',
+  apis: [],
+  path: '',
+};
+
+const defaultMetadata = {
+  site_name: 'Tech Docs',
+  site_description: 'Tech Docs',
+};
+
+const defaultEntity = {
+  kind: 'Component',
+  metadata: { namespace: 'default', name: 'docs' },
+};
+
+const defaultDom = (
+  <html lang="en">
+    <head />
+    <body>
+      <div data-md-component="container">
+        <div data-md-component="navigation" />
+        <div data-md-component="toc" />
+        <div data-md-component="main" />
+      </div>
+    </body>
+  </html>
+);
+
+/**
+ * test utility that can be used to build addons in techdocs page
+ * @public
+ */
+export class TechDocsAddonBuilder {
+  private options: TechDocsAddonBuilderOptions = defaultOptions;
+  private addons: ReactElement[];
+
+  static buildAddonsInTechDocs(addons: ReactElement[]) {
+    return new TechDocsAddonBuilder(addons);
+  }
+
+  constructor(addons: ReactElement[]) {
+    this.addons = addons;
+  }
+
+  withApis(apis: Apis) {
+    const refs = apis.map(([ref]) => ref);
+    this.options.apis = this.options.apis
+      .filter(([ref]) => !refs.includes(ref))
+      .concat(apis);
+    return this;
+  }
+
+  withDom(dom: ReactElement) {
+    this.options.dom = dom;
+    return this;
+  }
+
+  withMetadata(metadata: any) {
+    this.options.metadata = metadata;
+    return this;
+  }
+
+  withEntity(entity: any) {
+    this.options.entity = entity;
+    return this;
+  }
+
+  withReaderPage(readerPage: ReactElement) {
```

Comment:
We can skip this for now as we will only test the MkDocs based reader page currently